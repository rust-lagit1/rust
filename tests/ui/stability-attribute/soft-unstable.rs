// Test handling of soft-unstable items dependent on multiple features.
//@ aux-build:soft-unstable.rs
//@ revisions: all none
//@ [all]check-pass

#![cfg_attr(all, feature(a, b, c, d))]

extern crate soft_unstable;

fn main() {
    soft_unstable::mac!();
    //[none]~^ ERROR use of unstable library feature `a` [soft_unstable]
    //[none]~| WARNING this was previously accepted by the compiler but is being phased out
    //[none]~| ERROR use of unstable library feature `b`: reason b [soft_unstable]
    //[none]~| WARNING this was previously accepted by the compiler but is being phased out
    soft_unstable::something();
    //[none]~^ ERROR use of unstable library feature `c` [soft_unstable]
    //[none]~| WARNING this was previously accepted by the compiler but is being phased out
    //[none]~| ERROR use of unstable library feature `d`: reason d [soft_unstable]
    //[none]~| WARNING this was previously accepted by the compiler but is being phased out
}
