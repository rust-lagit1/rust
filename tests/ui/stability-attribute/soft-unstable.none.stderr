error: use of unstable library feature 'a'
  --> $DIR/soft-unstable.rs:11:5
   |
LL |     soft_unstable::mac!();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
   = note: `#[deny(soft_unstable)]` on by default

error: use of unstable library feature 'b': reason b
  --> $DIR/soft-unstable.rs:11:5
   |
LL |     soft_unstable::mac!();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>

error: use of unstable library feature 'c'
  --> $DIR/soft-unstable.rs:16:5
   |
LL |     soft_unstable::something();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>

error: use of unstable library feature 'd': reason d
  --> $DIR/soft-unstable.rs:16:5
   |
LL |     soft_unstable::something();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>

error: aborting due to 4 previous errors

Future incompatibility report: Future breakage diagnostic:
error: use of unstable library feature 'a'
  --> $DIR/soft-unstable.rs:11:5
   |
LL |     soft_unstable::mac!();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
   = note: `#[deny(soft_unstable)]` on by default

Future breakage diagnostic:
error: use of unstable library feature 'b': reason b
  --> $DIR/soft-unstable.rs:11:5
   |
LL |     soft_unstable::mac!();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
   = note: `#[deny(soft_unstable)]` on by default

Future breakage diagnostic:
error: use of unstable library feature 'c'
  --> $DIR/soft-unstable.rs:16:5
   |
LL |     soft_unstable::something();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
   = note: `#[deny(soft_unstable)]` on by default

Future breakage diagnostic:
error: use of unstable library feature 'd': reason d
  --> $DIR/soft-unstable.rs:16:5
   |
LL |     soft_unstable::something();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
   = note: `#[deny(soft_unstable)]` on by default

